int32 cursor_loc
int32 image_width
